webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js

```diff
@@ -286,8 +286,13 @@ export class SakaiRubric extends RubricsElement {
     this.rubric.weighted = !this.rubric.weighted;
     if (this.rubric.weighted) {
       this.rubric.criteria.forEach(cr => cr.weight = 0);
-      this.rubric.criteria[0].weight = 100;
-      this.handleSaveWeights(e);
+      //Try to get first criterion, that is not a criterion group
+      const fistCriterion = this.rubric.criteria.find(criteria => criteria.ratings?.length > 0);
```

Comment:
"firstCriterion"

Reply:
Optional chaining.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
You can compare undefined to a number safely, too. The comparison will always return false..